demo/chaincode/golang/auction.go

```diff
@@ -0,0 +1,265 @@
+		status := &Status{-1, err.Error()}
+		return shim.Success(buildReturnJson(status, nil))
+	}
+
+	status := &Status{0, "OK"}
+	return shim.Success(buildReturnJson(status, auctionStatus))
+}
+
+func submitClockBid(stub shim.ChaincodeStubInterface, args []string) pb.Response {
+	bidId := "1"
+	err := stub.PutState(bidKeyPrefix+bidId, []byte(args[0]))
+	if err != nil {
+		return shim.Error("PutState error: " + err.Error())
+	}
+
+	status := &Status{0, "OK"}
+	return shim.Success(buildReturnJson(status, nil))
+}
+
+func endRound(stub shim.ChaincodeStubInterface, args []string) pb.Response {
+	var request AuctionRequest
+	err := json.Unmarshal([]byte(args[0]), &request)
+	if err != nil {
+		return shim.Error("json error: " + err.Error())
+	}
+
+	auctionStatus, err := fetchAuctionStatus(stub, request.AuctionId)
+	if err != nil {
+		status := &Status{-1, err.Error()}
+		return shim.Success(buildReturnJson(status, nil))
+	}
+
+	auctionStatus.RoundActive = false
+
+	err = storeAuctionStatus(stub, request.AuctionId, auctionStatus)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+
+	status := &Status{0, "OK"}
+	return shim.Success(buildReturnJson(status, nil))
+}
+
+func startNextRound(stub shim.ChaincodeStubInterface, args []string) pb.Response {
+	var request AuctionRequest
+	err := json.Unmarshal([]byte(args[0]), &request)
+	if err != nil {
+		return shim.Error("json error: " + err.Error())
+	}
+
+	auctionStatus, err := fetchAuctionStatus(stub, request.AuctionId)
+	if err != nil {
+		status := &Status{-1, err.Error()}
+		// TODO discuss if this is a shim error os success?
+		return shim.Success(buildReturnJson(status, nil))
+	}
+
+	auctionStatus.ClockRound = auctionStatus.ClockRound + 1
+	auctionStatus.RoundActive = true
+
+	err = storeAuctionStatus(stub, request.AuctionId, auctionStatus)
+	if err != nil {
+		return shim.Error(err.Error())
+	}
+
+	status := &Status{0, "OK"}
+	return shim.Success(buildReturnJson(status, nil))
+}
+
+func getRoundInfo(stub shim.ChaincodeStubInterface, args []string) pb.Response {
+	var request RoundRequest
+	err := json.Unmarshal([]byte(args[0]), &request)
+	if err != nil {
+		return shim.Error("json error: " + err.Error())
+	}
+
+	auction, err := fetchAuctionDetails(stub, request.AuctionId)
+	if err != nil {
+		status := &Status{-1, err.Error()}
+		return shim.Success(buildReturnJson(status, nil))
+	}
+
+	// gather information for round
+	auctionStatus, err := fetchAuctionStatus(stub, request.AuctionId)
+	if err != nil {
+		status := &Status{-1, err.Error()}
+		// TODO discuss if this is a shim error os success?
+		return shim.Success(buildReturnJson(status, nil))
+	}
+
+	var prices []Price
+	for _, territory := range auction.Territories {
+		prices = append(prices, Price{
+			TerritoryId: territory.Id,
+			MinPrice:    territory.MinPrice,
+			ClockPrice:  int(float64(territory.MinPrice) * (1 + float64(auction.ClockPriceIncrementPercentage)/100.0)),
+		})
+	}
+
+	response := &RoundInfo{
+		Princes: prices,
```

Comment:
If you make the change in messages.go
```suggestion
		Prices: prices,
```

Reply:
fixed